Issue #46. Loops should not contain more than a single 'break' or 'continue' statement

sabaka opened the pull request:
Restricting the number of break and continue statements in a loop is done in the interest of good structured programming.

One break and continue statement is acceptable in a loop, since it facilitates optimal coding. If there is more than one, the code should be refactored to increase readability.

romani commented:
> [ERROR] com.puppycrawl.tools.checkstyle.checks.coding.AbstractSuperCheck failed coverage check. Branch coverage rate of 97.8% is below 100.0%

please do run "mvn clean verify" each time before "git push".

romani commented:
1) AbstractSuperCheck 

move that change to separate PR , and let other fix be applied. Do not block other changes.
Please investigate why that expression was covered 100% in one place and not covered in other place. That could be nuance/problem of cobertura. So please extend logging and prove that expression is passing with all value combinations.

2) AvoidStaticImportCheck.java

I expected only scope of "//this section allows explicit imports" to be moved to separate method, please explain.

sabaka commented:
Should be fixed. I did it again and understand, that I replaced inversion in the wrong way.

romani commented:
I do not see DONE for each my point, please do

sabaka commented:
1) fixed
2) I did it because both parts of if statements check whether current line is excluded. Actually, if I moved only second part I would have to add another break in second part of condition block to exit from the loop when dot found.

romani commented:
Please rebase your commit on latest sources and do "mvn clean verify" before push

romani commented:
please rebase on latest code, I fixed that problem #2065 .

romani commented:
commit message should be by format "Issue #XXXXX: ................"

romani commented:
merged as FF
